public/pages/workflow_detail/agentic_search/configure_flow/agent_mcp_servers.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { isEmpty } from 'lodash';
+import { getIn } from 'formik';
+import {
+  EuiAccordion,
+  EuiSpacer,
+  EuiButtonIcon,
+  EuiText,
+  EuiFormRow,
+  EuiSmallButtonEmpty,
+  EuiPanel,
+  EuiToolTip,
+  EuiIcon,
+  EuiLink,
+} from '@elastic/eui';
+import {
+  Agent,
+  DEFAULT_MCP_SERVER,
+  MCP_AGENT_CONFIG_DOCS_LINK,
+  MCPConnector,
+} from '../../../../../common';
+import { AppState } from '../../../../store';
+import { MCPToolFilters } from './mcp_tool_filters';
+import { MCPServerSelector } from './mcp_server_selector';
+
+interface AgentMCPServersProps {
+  agentForm: Partial<Agent>;
+  setAgentForm: (agentForm: Partial<Agent>) => void;
+}
+
+export function AgentMCPServers({
+  agentForm,
+  setAgentForm,
+}: AgentMCPServersProps) {
+  const { connectors } = useSelector((state: AppState) => state.ml);
+
+  // Persist state for each search MCP server accordion. Only support one at a time
+  const [openAccordionIndex, setOpenAccordionIndex] = useState<
```

Comment:
Should we use an array here?

Reply:
The idea was to just support one at a time to limit the vertical width and prevent multiple open accordions, and keep users focused on the specific accordion they're in. It's same logic as the search templates accordion.

I found that there looks to be a case where multiple accordions can still be opened, so I've done some more testing and fixed for both cases. LMK what you think.